Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -34,12 +40,55 @@ pub use num::ToPrimitive;
 pub use num::Zero;
 pub use std::convert::Into;
 
+pub use ::std::any::Any;
+pub use ::std::marker::Send;
+pub use ::std::marker::Sync;
+
+#[cfg(not(feature = "sync"))]
+pub type DynAny = dyn Any;
+#[cfg(feature = "sync")]
+pub type DynAny = dyn Any + Send + Sync;
+
+#[cfg(not(feature = "sync"))]
+pub use ::std::cell::UnsafeCell;
+
+#[cfg(feature = "sync")]
+pub struct UnsafeCell<T: ?Sized> {
+    data: ::std::cell::UnsafeCell<T>, // UnsafeCell for interior mutability
+}
+#[cfg(feature = "sync")]
+unsafe impl<T: ?Sized> Sync for UnsafeCell<T> where T: Send {}
```

Comment:
Why is unsafe necessary here? We should at least document it

Reply:
Documented + removed use of UnsafeCell for sequences to prefer the mutex just in case, and added tests.